Build failure: python3.12-pyliblo-0.10.0

## Steps To Reproduce

Steps to reproduce the behavior:

1. build pythonPackages312.pyliblo on nixos-unstable

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/7wqyb1sjxfchi52zw4miwbi6nv45a2w0-pyliblo-0.10.0.tar.gz
source root is pyliblo-0.10.0
setting SOURCE_DATE_EPOCH to timestamp 1428998296 of file pyliblo-0.10.0/setup.cfg
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/6j6v9hw90swd84a1jzr207f4r6al2bc0-py3.11.patch?id=a7e1eca5533657ddd7e37c43e67e8126e3447258
patching file src/liblo.pyx
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing setuptoolsBuildPhase
/nix/store/sfifx7g499n2xfvs5hzvw3mxinn0i17x-python3.12-setuptools-75.3.0/lib/python3.12/site-packages/setuptools/_distutils/d>
  warnings.warn(msg)
running bdist_wheel
running build
running build_ext
cythoning src/liblo.pyx to src/liblo.c
/nix/store/1h9jkn3m4nnpi1pg5i77czinqnzqrpa7-python3.12-cython-0.29.37.1/lib/python3.12/site-packages/Cython/Compiler/Main.py:>
  tree = Parsing.p_module(s, pxd, full_module_name)
building 'liblo' extension
creating build/temp.linux-x86_64-cpython-312/src
gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -I/nix/store/6cr0spsvymmrp1hj5n0kbaxw55w1lqyp-libxcrypt-4.4.36/>
src/liblo.c: In function ‘__pyx_f_5liblo__msg_callback’:
src/liblo.c:4749:94: error: passing argument 1 of ‘lo_blob_dataptr’ from incompatible pointer type [-Wincompatible-pointer-ty>
 4749 |         __pyx_t_6 = __Pyx_PyBytes_FromCString(((unsigned char *)lo_blob_dataptr((__pyx_v_argv[__pyx_v_i])))); if (unl>
      |                                                                                 ~~~~~~~~~~~~~^~~~~~~~~~~~
      |                                                                                              |
      |                                                                                              lo_arg *
src/liblo.c:831:78: note: in definition of macro ‘__Pyx_PyBytes_FromCString’
  831 | #define __Pyx_PyBytes_FromCString(s)   __Pyx_PyBytes_FromString((const char*)s)
      |                                                                              ^
compilation terminated due to -Wfatal-errors.
error: command '/nix/store/xcn9p4xxfbvlkpah7pwchpav4ab9d135-gcc-wrapper-14-20241116/bin/gcc' failed with exit code 1
```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

Error occurred during system rebuild as a dependency of zrythm

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.5, NixOS, 25.05 (Warbler), 25.05.20241217.d3c42f1`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `""`
 - nixpkgs: `/nix/store/6zgbbqlr7nnfxpzkyj7fsl4fpg89jbw0-source`

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@Sigmanificient (as there are no maintainers listed and you were the last person to touch the package)
@vcunat (issue seems to be related to gcc: see error)

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/pyliblo/default.nix
+++ b/pkgs/development/python-modules/pyliblo/default.nix
@@ -23,6 +23,11 @@ buildPythonPackage rec {
       url = "https://git.alpinelinux.org/aports/plain/community/py3-pyliblo/py3.11.patch?id=a7e1eca5533657ddd7e37c43e67e8126e3447258";
       hash = "sha256-4yCWNQaE/9FHGTVuvNEimBNuViWZ9aSJMcpTOP0fnM0=";
     })
+    # Fix compile error due to  incompatible pointer type 'lo_blob_dataptr'
+    (fetchurl {
+      url = "https://github.com/dsacre/pyliblo/commit/ebbb255d6a73384ec2560047eab236660d4589db.patch?full_index=1";
+      hash = "sha256-ZBAmBxSUT2xgoDVqSjq8TxW2jz3xR/pdCf2O3wMKvls=";
+    })
   ];
 
   build-system = [ cython_0 ];

